app/helpers/arclight_helper.rb

```diff
@@ -43,6 +43,14 @@ def fields_have_content?(document, field_accessor)
     end
   end
 
+  # If we have a facet on the repository, then return the Repository object for it
+  #
+  # @return [Repository]
+  def repository_faceted_on
+    repos = facets_from_request.find { |f| f.name == 'repository_sim' }.try(:items)
+    Arclight::Repository.find_by(name: repos.first.value) if repos.length == 1 && repos.first.value
```

Comment:
I can add that to the guard. I think it'd only happen if the repository facet was empty though, right? Maybe there's a smarter way to detect if we're faceted by repository?